### PR TITLE
Build: ge.gradle add tags & projectId=solr

### PR DESCRIPTION
**Pull request by dsmiley**
Add more Gradle Enterprise build metadata.

(replacing previous #2256 )

### CODE REVIEW

**Comment by dsmiley** [comment]
image
And the build details: https://ge.apache.org/s/7tfc5bq6jabyi

File: gradle/ge.gradle [hunk 1]
```diff
@@ -27,6 +31,25 @@ gradleEnterprise {
         obfuscation {
             ipAddresses { addresses -> addresses.collect { address -> "0.0.0.0"} }
         }
+
+        tag("JDK" + System.getProperty("java.version").find(/\d+/)) // major version
```
Comment (clayburn) [inline gradle/ge.gradle:35]: For what it's worth, these JDK details are already captured in the [build scan](https://ge.apache.org/s/iria4ppalnbdo#infrastructure). This information will show up in build scan comparisons as well. But capturing as tags and value is nice if you want the ability to filter on these values.

File: gradle/ge.gradle [hunk 2]
```diff
@@ -27,6 +31,25 @@ gradleEnterprise {
...
+            tag(jenkinsJobName.replaceAll(/Solr\W/,"")) // like "Smoketest-9.5"
+        }
+
+        // https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+        if (System.getenv("GITHUB_BASE_REF")) { // only exists for PRs
+            tag("PR")
+        }
+        String ghWorkflowName = System.getenv("GITHUB_WORKFLOW")
+        if (ghWorkflowName) {
+            tag(ghWorkflowName)
```
Comment (clayburn) [inline gradle/ge.gradle:51]: The Common Custom User Data Gradle Plugin already adds this as a custom value called `CI workflow`

File: gradle/ge.gradle [hunk 3]
```diff
@@ -27,6 +31,25 @@ gradleEnterprise {
...
         }
+
+        tag("JDK" + System.getProperty("java.version").find(/\d+/)) // major version
+        value("Java Vendor", System.getProperty("java.vendor"))
+        value("Java Version", System.getProperty("java.version"))
+
+        // Jenkins job name, less redundant "Solr" parts
+        String jenkinsJobName = System.getenv("JOB_NAME")
+        if (jenkinsJobName) { // looks like "Solr/Solr-Smoketest-9.5"
+            tag(jenkinsJobName.replaceAll(/Solr\W/,"")) // like "Smoketest-9.5"
```
Comment (clayburn) [inline gradle/ge.gradle:42]: The Common Custom User Data Gradle Plugin already adds this as a custom value called `CI job` (but without the filtering you've added).

**Review by clayburn — COMMENTED** [review]
This all looks great, but I added a few comments just so you are aware of what already is available.

File: gradle/ge.gradle (hunk 1 above)
Comment (dsmiley) [reply gradle/ge.gradle:35]: Yes; this is exactly why I added it.  It's a major piece of useful information that differentiates builds.

File: gradle/ge.gradle (hunk 2 above)
Comment (dsmiley) [reply gradle/ge.gradle:51]: Yes but I want a tag.  The Jenkins Job Name & GitHub workflow (similar) is a major build scan differentiator to quickly know what we're looking at.

File: gradle/ge.gradle (hunk 3 above)
Comment (dsmiley) [reply gradle/ge.gradle:42]: Wanted to keep tags short

**Pull request merged**